### PR TITLE
[swift] Always conform parent struct type to ProtoExtensible

### PR DESCRIPTION
**Pull request by dnkoutso**

### CODE REVIEW

File: wire-tests-swift/no-manifest/src/main/swift/AllTypes.swift [hunk 1]
```diff
@@ -848,6 +848,9 @@ public struct AllTypes {
 
 }
 
+extension AllTypes : ProtoExtensible {
```
Comment (lickel) [inline wire-tests-swift/no-manifest/src/main/swift/AllTypes.swift:851]: This also should be generating:
```
public var ext_opt_int32: Int32? {]
    get { storage.ext_opt_int32 }
    set { storage.ext_opt_int32 = newValue }
}
```

Comment (lickel) [reply wire-tests-swift/no-manifest/src/main/swift/AllTypes.swift:851]: Otherwise you'd be paying `dynamicMemberLookup` costs

Comment (dnkoutso) [reply wire-tests-swift/no-manifest/src/main/swift/AllTypes.swift:851]: Will be updating.

File: wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt [hunk 2]
```diff
@@ -1177,10 +1178,71 @@ class SwiftGenerator private constructor(
     )
   }
 
+  private fun TypeSpec.Builder.generateMessageExtensionStorageDelegates(
+    type: MessageType,
+    storageName: String,
+    structType: DeclaredTypeName,
+    fileMembers: MutableList<FileMemberSpec>,
+  ) {
+    if (!type.isHeapAllocated) {
```
Comment (dnkoutso) [inline wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt:1187]: copy pasted warning pattern from other methods that do similar stuff.

File: wire-tests-swift/no-manifest/src/main/swift/AllTypes.swift (hunk 1 above)
Comment (dnkoutso) [reply wire-tests-swift/no-manifest/src/main/swift/AllTypes.swift]: done

File: wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt [hunk 3]
```diff
@@ -1192,17 +1254,21 @@ class SwiftGenerator private constructor(
...
+                if (!forStorageType && field.documentation.isNotBlank()) {
                   addDoc("\n%L\n", field.documentation.sanitizeDoc())
                 }
-                addDoc("\nSource: %L\n", field.location.withPathOnly())
+                if (!forStorageType) {
+                  addDoc("\nSource: %L\n", field.location.withPathOnly())
+                }
 
-                if (field.isDeprecated) {
+                if (!forStorageType && field.isDeprecated) {
```
Comment (lickel) [inline wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt:1271]: I _think_ we want to mark as deprecated regardless?

Comment (dnkoutso) [reply wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt:1271]: we are in the other method for generating storage delegates?

**Review by lickel — APPROVED** [review]

File: wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt (hunk 3 above)
Comment (dnkoutso) [reply wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt]: nm spoke offline we want in both and updated.

**Pull request merged**